src/yxy.rs

```diff
@@ -0,0 +1,301 @@
+        }
+    }
+}
+
+impl<T: Float> Shade for Yxy<T> {
+    type Scalar = T;
+
+    fn lighten(&self, amount: T) -> Yxy<T> {
+        Yxy {
+            x: self.x,
+            y: self.y,
+            y_luma: self.y_luma + amount,
+        }
+    }
+}
+
+impl<T: Float> Default for Yxy<T> {
+    fn default() -> Yxy<T> {
+        Yxy::new(T::zero(), T::zero(), T::zero())
+    }
+}
+
+impl<T: Float> Add<Yxy<T>> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn add(self, other: Yxy<T>) -> Yxy<T> {
+        Yxy {
+            x: self.x + other.x,
+            y: self.y + other.y,
+            y_luma: self.y_luma + other.y_luma,
+        }
+    }
+}
+
+impl<T: Float> Add<T> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn add(self, c: T) -> Yxy<T> {
+        Yxy {
+            x: self.x + c,
+            y: self.y + c,
+            y_luma: self.y_luma + c,
+        }
+    }
+}
+
+impl<T: Float> Sub<Yxy<T>> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn sub(self, other: Yxy<T>) -> Yxy<T> {
+        Yxy {
+            x: self.x - other.x,
+            y: self.y - other.y,
+            y_luma: self.y_luma - other.y_luma,
+        }
+    }
+}
+
+impl<T: Float> Sub<T> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn sub(self, c: T) -> Yxy<T> {
+        Yxy {
+            x: self.x - c,
+            y: self.y - c,
+            y_luma: self.y_luma - c,
+        }
+    }
+}
+
+impl<T: Float> Mul<Yxy<T>> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn mul(self, other: Yxy<T>) -> Yxy<T> {
+        Yxy {
+            x: self.x * other.x,
+            y: self.y * other.y,
+            y_luma: self.y_luma * other.y_luma,
+        }
+    }
+}
+
+impl<T: Float> Mul<T> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn mul(self, c: T) -> Yxy<T> {
+        Yxy {
+            x: self.x * c,
+            y: self.y * c,
+            y_luma: self.y_luma * c,
+        }
+    }
+}
+
+impl<T: Float> Div<Yxy<T>> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn div(self, other: Yxy<T>) -> Yxy<T> {
+        Yxy {
+            x: self.x / other.x,
+            y: self.y / other.y,
+            y_luma: self.y_luma / other.y_luma,
+        }
+    }
+}
+
+impl<T: Float> Div<T> for Yxy<T> {
+    type Output = Yxy<T>;
+
+    fn div(self, c: T) -> Yxy<T> {
+        Yxy {
+            x: self.x / c,
+            y: self.y / c,
+            y_luma: self.y_luma / c,
+        }
+    }
+}
+
+from_color!(to Yxy from Xyz, Rgb, Luma, Lab, Lch, Hsv, Hsl);
+
+alpha_from!(Yxy {Xyz, Rgb, Luma, Lab, Lch, Hsv, Hsl, Color});
+
+impl<T: Float> From<Xyz<T>> for Yxy<T> {
+    fn from(xyz: Xyz<T>) -> Yxy<T> {
+        Yxy {
+            x: xyz.x / ( xyz.x + xyz.y + xyz.z ),
+            y: xyz.y / ( xyz.x + xyz.y + xyz.z ),
+            y_luma: xyz.y,
+        }
+    }
+}
+
+impl<T: Float> From<Luma<T>> for Yxy<T> {
+    fn from(luma: Luma<T>) -> Yxy<T> {
+        Yxy {
+            x: T::zero(),
+            y: T::zero(),
```

Comment:
We are still assuming D65, so it should be the D65 white point. We should add tests for this...